lib/templatesimplifier.cpp

```diff
@@ -829,15 +859,15 @@ void TemplateSimplifier::simplifyTemplateAliases()
             for (const Token *tok = startToken; tok != endToken; tok = tok->next()) {
                 if (!Token::Match(tok, "%name% <"))
                     continue;
-                std::list<TokenAndName>::iterator it = std::find_if(mTemplateInstantiations.begin(),
+                std::vector<TokenAndName>::iterator it = std::find_if(mTemplateInstantiations.begin(),
                                                        mTemplateInstantiations.end(),
                                                        FindToken(tok));
                 if (it == mTemplateInstantiations.end())
                     continue;
-                std::list<TokenAndName>::iterator next = it;
+                std::vector<TokenAndName>::iterator next = it;
                 ++next;
-                if (it == it1)
-                    it1 = next;
+                if (*it == mTemplateInstantiations[i])
```

Comment:
this will perform a value comparison as far as I see. maybe that is not expensive but I guess we could just match the address instead?